product_types/data_tabulation/sdtm.py

```diff
@@ -44,23 +44,26 @@ def get_metadata(self, parent: str = None) -> ([dict], [dict], [dict]):
 
         # link variables to appropriate parent structure
         for variable in variables:
-            parent_dataset_name = variable.parent_dataset_name.replace("SDTM ", "").replace("SEND ", "")
-            parent_class_name = variable.parent_class_name.replace("SDTM ", "").replace("SEND ", "")
-            parent_dataset = self._find_dataset(parent_dataset_name, datasets)
-            parent_class = self._find_class_by_name(parent_class_name, classes)
-            if variable.variables_qualified:
-                self._add_qualified_variables_link(variable, variables)
-            if self.is_ig:
-                if parent_class and parent_class.parent_class_name:
-                    variable.build_model_class_variable_link()
-                else:
-                    variable.build_model_dataset_variable_link()
-            if parent_dataset:
-                variable.set_parent_dataset(parent_dataset)
-                parent_dataset.add_variable(variable)
-            elif not self.is_ig and parent_class:
-                variable.set_parent_class(parent_class)
-                parent_class.add_variable(variable)
+            try:
+                parent_dataset_name = variable.parent_dataset_name.replace("SDTM ", "").replace("SEND ", "")
+                parent_class_name = variable.parent_class_name.replace("SDTM ", "").replace("SEND ", "")
+                parent_dataset = self._find_dataset(parent_dataset_name, datasets)
+                parent_class = self._find_class_by_name(parent_class_name, classes)
+                if variable.variables_qualified:
+                    self._add_qualified_variables_link(variable, variables)
+                if self.is_ig:
+                    if parent_class and parent_class.parent_class_name:
+                        variable.build_model_class_variable_link()
+                    else:
+                        variable.build_model_dataset_variable_link()
+                if parent_dataset:
+                    variable.set_parent_dataset(parent_dataset)
+                    parent_dataset.add_variable(variable)
+                elif not self.is_ig and parent_class:
+                    variable.set_parent_class(parent_class)
+                    parent_class.add_variable(variable)
+            except:
+                print("Error for variable with label: ", variable.label)
```

Comment:
I think instead of a try-except, we should check whether the `parent_dataset_name` is `None`.
If it is `None`, we should continue with `parent_dataset_name = ""` instead of `None` and then use `logger.error` instead of `print`